### ISSUE TITLE
`npm audit fix`

### ISSUE THREAD

**Issue by AurelienLourot** [Frameright/image-display-control-web-component#10]
At the moment, NPM detects a high severity vulnerability in one of our dependency. However `npm audit fix` doesn't help:

```
$ npm audit fix

up to date, audited 774 packages in 3s

133 packages are looking for funding
  run `npm fund` for details

# npm audit report

marked  <=4.0.9
Severity: high
Marked ReDoS due to email addresses being evaluated in quadratic time - https://github.com/advisories/GHSA-xf5p-87ch-gxw2
Inefficient Regular Expression Complexity in marked - https://github.com/advisories/GHSA-5v2h-r2cx-5xgj
Inefficient Regular Expression Complexity in marked - https://github.com/advisories/GHSA-rrrm-qjm4-v8hf
fix available via `npm audit fix`
node_modules/marked

1 high severity vulnerability

To address all issues, run:
  npm audit fix
```

**Comment by MarkovianPD** [comment]
I would like to take a crack at this!

**Comment by AurelienLourot** [comment]
Thank you @MarkovianPD !

**Comment by MarkovianPD** [comment]
> Thank you @MarkovianPD !

Absolutely! I'm new to contributions trying to get my foot in the door anyway possible and help out! I created a fork to mess around with it before your reply and from what I could dig up and conclude is that you'll want to upgrade npm package "marked" to version 0.6.2. I'm not sure if I can do that by editing the package.json version to the updated version and it will upgrade? It does also seem you're using a 0.3.X version and I assume you're using an older version for a reason and upgrading may not be the best solution? Any and all feedback would be appreciated!

**Comment by AurelienLourot** [comment]
`marked` is not one of our direct dependencies: https://github.com/Frameright/image-display-control-web-component/blob/main/image-display-control/package.json#L44

It must be a dependency of one of all these dependencies. So if we're using an old version, it's not on purpose from us, but it might be on purpose from one of these dependencies, I'm not sure. If pinning explicitly the latest `marked` in our `package.json` works, I'm happy with that workaround :) Thanks!

**Comment by MarkovianPD** [comment]
Very interesting! If you could explained the process of pinning the latest in your package.json I would be thankful and more than happy to get it done asap!

**Comment by MarkovianPD** [comment]
This is the only alternative I could find, https://socket.dev/npm/package/md-spell it says it is loosely based on lukeapage's node-markdown. I'm not sure this would accomplish exactly what you're looking for. Pertaining to the ugprade some users experiencing this issue in the past have commented that running npm upgrade solved the issue I would recommend at least trying this but as mentioned earlier there may be some issues if you're not wanting to upgrade specific packages. Multiple users responded to the recommendation stating that had worked for their situation.

**Comment by AurelienLourot** [comment]
`npm upgrade` doesn't help in my case. I think this is because markdown-spellcheck pins `^0.7.0`: https://github.com/lukeapage/node-markdown-spellcheck/blob/master/package.json#L40

Which means "any version between 0.7.0 and 1", so this prevents us from getting the latest `marked`.

https://www.npmjs.com/package/md-spell has no link to a github repository, so I'm not sure where the code lives. There is a Tab `Code` there and I can see in `package.json` that they don't use `marked` anymore, which is good. They seem to use `hunspell-spellchecker` as a dependency, which hasn't been published for 8 years: https://www.npmjs.com/package/hunspell-spellchecker . I fear we would be switching to another abandoned project and run into the same kind of issues

**Comment by MarkovianPD** [comment]
I see, great catch I believe you're correct about jumping to another abandoned project. I believe I don't have sufficient knowledge to assist further with this but thank you for letting me have the chance to help you! I hope I can assist with other contributions in the future. If you find anything more on this npm audit and I can help I'll be ecstatic to help out!

**Comment by MarkovianPD** [comment]
@ilu Thank you for chiming in! I'll take a look at it when I can and see if any of it makes sense haha I'd love to crack the case and get it solved!

**Comment by MarkovianPD** [comment]
Thank you all for the help trying to get to the bottom of this! I've taken a read up on the cspell blog post and I believe this is beyond my expertise at the moment.

**Comment by ilu** [comment]
> Thank you all for the help trying to get to the bottom of this! I've taken a read up on the cspell blog post and I believe this is beyond my expertise at the moment.

No worries @MarkovianPD, thank you for having a look. Please drop by again to see if any new issues would be suitable for you :)

**Comment by AurelienLourot** [comment]
FYI the spellchecker is now gone but we still have some `npm audit` issues. Keeping this open for now.

**Comment by AurelienLourot** [comment]
My mistake, I had forgotten to actually remove the dependency. Now the remaining `npm audit` issues are only `moderate`.
